provider/qdrant.go

```diff
@@ -0,0 +1,293 @@
+	pc "github.com/featureform/provider/provider_config"
+	pt "github.com/featureform/provider/provider_type"
+	"github.com/featureform/provider/types"
+	"github.com/google/uuid"
+	pb "github.com/qdrant/go-client/qdrant"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/metadata"
+)
+
+type qdrantOnlineStore struct {
+	collections_client pb.CollectionsClient
+	points_client      pb.PointsClient
+	service_client     pb.QdrantClient
+	connection         *grpc.ClientConn
+	apiKey             string
+	BaseProvider
+}
+
+func qdrantOnlineStoreFactory(serialized pc.SerializedConfig) (Provider, error) {
+	qdrantConfig := &pc.QdrantConfig{}
+	if err := qdrantConfig.Deserialize(serialized); err != nil {
+		return nil, err
+	}
+	return NewQdrantOnlineStore(qdrantConfig)
+}
+
+func NewQdrantOnlineStore(options *pc.QdrantConfig) (*qdrantOnlineStore, error) {
+
+	var tlsCredential credentials.TransportCredentials
+
+	if !options.UseTls && options.ApiKey != "" {
+		log.Println("Warning: API key is set but TLS is not enabled. The API key will be sent in plaintext.")
+		log.Println("May fail when using Qdrant cloud.")
+	}
+
+	if options.UseTls {
+		tlsCredential = credentials.NewTLS(&tls.Config{})
+	} else {
+		tlsCredential = insecure.NewCredentials()
+	}
+
+	conn, err := grpc.Dial(options.GrpcHost, grpc.WithTransportCredentials(tlsCredential))
+	if err != nil {
+		return nil, fferr.NewConnectionError(pt.QdrantOnline.String(), err)
+	}
+
+	return &qdrantOnlineStore{
+		collections_client: pb.NewCollectionsClient(conn),
+		points_client:      pb.NewPointsClient(conn),
+		service_client:     pb.NewQdrantClient(conn),
+		connection:         conn,
+		apiKey:             options.ApiKey,
+		BaseProvider: BaseProvider{
+			ProviderType:   pt.QdrantOnline,
+			ProviderConfig: options.Serialize(),
+		},
+	}, nil
+}
+
+func (store *qdrantOnlineStore) AsOnlineStore() (OnlineStore, error) {
+	return store, nil
+}
+
+func (store *qdrantOnlineStore) Close() error {
+	return store.connection.Close()
+}
+
+func (store *qdrantOnlineStore) CreateTable(feature, variant string, valueType types.ValueType) (OnlineStoreTable, error) {
+	return &qdrantOnlineTable{
+		store:          store,
+		collectionName: store.getCollectionName(feature, variant),
+		valueType:      valueType,
+	}, nil
+}
+
+func (store *qdrantOnlineStore) DeleteTable(feature, variant string) error {
+
+	ctx := store.appendApiKeyToContext(context.TODO())
+
+	_, err := store.collections_client.Delete(ctx, &pb.DeleteCollection{
+		CollectionName: store.getCollectionName(feature, variant),
+	})
+
+	return err
+}
+
+func (store *qdrantOnlineStore) CheckHealth() (bool, error) {
+
+	ctx := store.appendApiKeyToContext(context.TODO())
+
+	response, err := store.service_client.HealthCheck(ctx, &pb.HealthCheckRequest{})
+
+	if err != nil {
+		return false, err
+	}
+	return response.GetTitle() == "qdrant - vector search engine", nil
```

Comment:
Just quick sanity check, is this the best way to health check? Any chance of that title changing?

Reply:
Can be almost certain about this not changing.